pkg/collector/corechecks/servicediscovery/language/language.go

```diff
@@ -146,12 +150,21 @@ func FindInArgs(args []string) Language {
 	if len(args) == 0 {
 		return ""
 	}
-	for i := 0; i < len(args); i++ {
-		procName := path.Base(args[i])
-		// if procName is a known language, return the pos and the language
-		if lang, ok := procToLanguage[procName]; ok {
-			return lang
-		}
+
+	langs := languagedetection.DetectLanguage([]languagemodels.Process{&procutil.Process{
+		// Pid doesn't matter since sysprobeConfig is nil
+		Pid:     0,
+		Cmdline: args,
+		Comm:    args[0],
+	}}, nil)
+
+	lang := langs[0]
```

Comment:
is it safe to assume we always have at least one entry?

Reply:
Currently `DetectLanguage` always returns the correct number of entries, but I've added a check above for safety.